### DISCUSSION TITLE
Is It Possible to Store Only Thumbnails and Discard Original Images in PocketBase?

### DISCUSSION THREAD

**Discussion by shajidhasan** [pocketbase/pocketbase#6012] [Q&A] [answered]
I'm using PocketBase for my app and have a use case where I only need the generated thumbnail images for avatars. Storing the original image files is unnecessary for my application, and I'd like to save storage space by discarding them after the thumbnails are generated. Or replace the original file with one of the thumbnails.

Is this possible?

Or, can I just run something like `cp thumbs_[image]/100x100f_[image].jpg ./[image].jpg` in every directory, to replace the original image with a thumbnail, without breaking anything? This isn't optimal because there will be an unnecessary duplication, but it'll still help me.

Thank you so much.

**Comment by ganigeorgiev** [comment, marked as answer]
No, it is not possible because thumbs are generated "lazily" (aka. on first access) and relies on the existence of the original file.

An alternative could be to generate the thumb on the client-side and upload the generated image (_or eventually intercept the create/update request and modify the uploaded files by replacing them with server-side generated image before save but this is more involving_).

**Reply by shajidhasan** [reply]
Thank you so much for the reply!

I'll update my app to generate thumbs on device and then upload. But if run the `cp thumbs_[image]/100x100f_[image].jpg ./[image].jpg` trick for only the images with generated thumbnails, it should not break anything, right?

**Reply by ganigeorgiev** [reply]
I don't recommend it. Not only you'll store the thumb twice, you also can't reliably know when the thumb will be generated because it may have not been requested yet.
